### PR TITLE
restyle flow navigation

### PR DESCRIPTION
**Pull request by chohner**
Depends on https://github.com/digitalservicebund/a2j-rechtsantragstelle/pull/852

look:
image

Todo:
- [x] `DoneCurrent` state
- [x] restyle active parent
- [ ] potential refactor enum

### CODE REVIEW

File: app/services/navigation/navState.ts [hunk 1]
```diff
@@ -1,7 +1,27 @@
...
+  DoneCurrent,
   Current,
   Open,
   OpenDisabled,
 }
+
+export const stateIsCurrent = (state: NavState) =>
+  state === NavState.Current || state === NavState.DoneCurrent;
+
+export const stateIsActive = (state: NavState) =>
```
Comment (aaschlote) [inline app/services/navigation/navState.ts:13]: do we have an unit test for it?

File: app/services/navigation/navState.ts [hunk 2]
```diff
@@ -1,7 +1,27 @@
...
+  [
+    NavState.DoneCurrent,
+    NavState.Current,
+    NavState.Open,
+    NavState.Done,
+  ].includes(state);
+
+export const stateIsDone = (state: NavState) =>
```
Comment (aaschlote) [inline app/services/navigation/navState.ts:21]: do we have an unit test for it?

**Review by aaschlote — COMMENTED** [review]
Can we think about to break the FlowNavigation.tsx in small parts? I have the feeling it's very difficult to read it, because in this file has 4 react components and 3 types.

Idea: create a new folder called `flowNavigation` or `navigation` under the components, it would be something

components
    &emsp;flowNavigation or navigation
  &emsp;&emsp;StateIcon.tsx
  &emsp;&emsp;NavigationList.tsx
   &emsp;&emsp;FlowNavigation.tsx
   &emsp;&emsp;NavItem.tsx

**Comment by chohner** [comment]
I had similar ideas regarding refactoring but I wanted to keep this PR close to a restyle as possible.

File: app/services/navigation/navState.ts (hunk 1 above)
Comment (chohner) [reply app/services/navigation/navState.ts:13]: I would like to wait with unit test for two reasons: This is imo too trivial (dangerous statement, I know) and there is no way to unit test this without recreating the functionality of this helper. More importantly, I think we should refactor this to remove the `NavState` enum altogether (again, not in this PR)

File: app/components/FlowNavigation.tsx [hunk 3]
```diff
@@ -78,58 +60,52 @@ export default function FlowNavigation(props: FlowNavigationProps) {
...
+
+  const liClassNames = classNames("list-none", {
+    "text-gray-600 curser-not-allowed hover:font-normal pointer-events-none":
+      isDisabled,
+    "border-b-[1px] border-b-white last:border-b-0": !isChild,
+    "border-l-[1px] border-l-blue-800": isCurrent,
+  });
+
+  const itemClassNames = classNames(
+    "w-full	ds-label-02-reg p-16 flex gap-x-16 items-center hover:underline hover:bg-blue-300 active:bg-white	focus-visible:shadow-[inset_0px_0px_0px_4px] focus:shadow-blue-800",
```
Comment (pgurusinga) [inline app/components/FlowNavigation.tsx:88]: why they have big whitespace? probably tab?
```suggestion
    "w-full ds-label-02-reg p-16 flex gap-x-16 items-center hover:underline hover:bg-blue-300 active:bg-white focus-visible:shadow-[inset_0px_0px_0px_4px] focus:shadow-blue-800",
```

File: app/components/FlowNavigation.tsx [hunk 4]
```diff
@@ -78,58 +60,52 @@ export default function FlowNavigation(props: FlowNavigationProps) {
...
           <button
-            className={`${stateClassNames} md:pr-0 min-w-[242px] relative flex items-center w-full cursor-pointer flex gap-x-16 items-center`}
+            className={itemClassNames}
             aria-disabled={isDisabled}
             aria-expanded={collapse.isExpanded}
             {...collapse.getToggleProps()}
             aria-describedby={iconId}
           >
-            <StateIcon isDone={isDone} id={iconId} a11yLabels={a11yLabels} />
+            {isDone && <StateIcon id={iconId} a11yLabels={a11yLabels} />}
```
Comment (pgurusinga) [inline app/components/FlowNavigation.tsx:108]: this is a good one!

**Review by pgurusinga — APPROVED** [review]

**Pull request merged**